test/builtin_dataset_mocks.py

```diff
@@ -571,16 +590,22 @@ def generate(
         cls,
         root,
         *,
+        split,
         year,
         num_samples,
     ):
         annotations_dir = root / "annotations"
         annotations_dir.mkdir()
 
-        for split in ("train", "val"):
-            config_name = f"{split}{year}"
+        for split_ in ("train", "val"):
+            config_name = f"{split_}{year}"
+
+            images_meta = cls._make_images_folder(root, config_name, num_samples=num_samples)
+            if split_ == split:
+                make_zip(root, f"{config_name}.zip")
+            else:
+                shutil.rmtree(root / config_name)
```

Comment:
Could you please explain what this does and why this is needed?

Reply:
Before this patch, the mock data generation always generated the train and the validation archive. Since only one of them is ever used by a given config, we delete the one that is not used. Instead of doing that, we could also generate the image metadata manually and only create an image folder for the correct split, rather than generate both and extract the metadata from it.